flowey/flowey_core/src/pipeline.rs

```diff
@@ -947,6 +948,20 @@ impl PipelineJob<'_> {
         self
     }
 
+    /// Overrides the id of the job.
+    ///
+    /// Flowey typically generates a reasonable job ID but some use cases that depend
+    /// on the ID may find it useful to override it to something custom.
+    pub fn ado_override_job_id<S>(self, name: S) -> Self
+    where
+        S: AsRef<str>,
+    {
+        self.pipeline
+            .ado_job_name_overrides
```

Comment:
you'll want to percolate the swap from `name` to `id` terminology down into the internals as well (to avoid confusion for future folks looking at this code)